### PR TITLE
New data source: 'azurerm_eventgrid_topic'

### PR DESCRIPTION
**Pull request by brunhil**
Expose the 'azurerm_eventgrid_topic' resource as a data source

### CODE REVIEW

File: azurerm/internal/services/eventgrid/data_source_eventgrid_topic.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```
Comment (katbyte) [inline azurerm/internal/services/eventgrid/data_source_eventgrid_topic.go]: Could we validate this?

File: azurerm/internal/services/eventgrid/data_source_eventgrid_topic.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"tags": tags.Schema(),
```
Comment (katbyte) [inline azurerm/internal/services/eventgrid/data_source_eventgrid_topic.go]: Could we put this as the last property?

**Comment by brunhil** [comment]
Fixes pushed and verified tests run! 

Event Grid Topic Data Source
```
=== RUN   TestAccDataSourceAzureRMEventGridTopic_basic
=== PAUSE TestAccDataSourceAzureRMEventGridTopic_basic
=== CONT  TestAccDataSourceAzureRMEventGridTopic_basic
--- PASS: TestAccDataSourceAzureRMEventGridTopic_basic (245.73s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/eventgrid/tests     246.001s
```

Event Grid Topic Resource
```
=== RUN   TestAccAzureRMEventGridTopic_basic
=== PAUSE TestAccAzureRMEventGridTopic_basic
=== RUN   TestAccAzureRMEventGridTopic_basicWithTags
=== PAUSE TestAccAzureRMEventGridTopic_basicWithTags
=== CONT  TestAccAzureRMEventGridTopic_basic
=== CONT  TestAccAzureRMEventGridTopic_basicWithTags
--- PASS: TestAccAzureRMEventGridTopic_basic (153.90s)
--- PASS: TestAccAzureRMEventGridTopic_basicWithTags (196.36s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/eventgrid/tests     196.684s
```

File: azurerm/internal/services/eventgrid/data_source_eventgrid_topic.go (hunk 1 above)
Comment (brunhil) [reply azurerm/internal/services/eventgrid/data_source_eventgrid_topic.go]: Added the **NoEmptyStrings** validation to this and the **resource_azurerm_eventgrid_topic** resource. Let me know if you had a different type of validation in mind.

File: website/docs/d/eventgrid_topic.html.markdown [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Data Source: azurerm_eventgrid_topic
+
+Use this data source to access information about an existing EventGrid Topic
+
+## Example Usage
+
+```hcl
+data "azurerm_eventgrid_topic" "example" {
+  name                = "my-eventgrid-topic"
+  resource_group_name = "${azurerm_resource_group.example.name}"
```
Comment (tombuildsstuff) [inline website/docs/d/eventgrid_topic.html.markdown]: can we make this:

```suggestion
  resource_group_name = "example-resources"
```

**Pull request merged**